app/segments/(team)/chooseTeam.js

```diff
@@ -99,22 +104,56 @@ function ChooseTeam() {
           >
             <Button
               onPress={async () => {
-                //temporary, should be replaced with multiple team functionality
-                await setDoc(doc(db, "teams", "1", "users", currentUserId), {
-                  name: currentUserInfo["displayName"],
-                  // hardcoded pfp string for now, add pfp upload to profile settings in future PR
-                  pfp: "",
-                  // hardcoded "player" role for now, add role selection to profile settings in future PR
-                  role: "player",
-                  uid: currentUserId,
-                  assigned_data: [],
-                  uniqueDrills: [],
-                });
-                setCurrentUserId(currentUserId);
-                await invalidateMultipleKeys(queryClient, [
-                  ["userInfo", { userId: currentUserId }],
-                ]);
-                router.replace("/");
+                let completed = false;
+
+                const unregisterAuthObserver = onIdTokenChanged(
```

Comment:
Nice job finding out a different observer for this! Shouldn't this observer be registered when this page is loaded?

Reply:
Is the best way to do that to put the code in a useEffect or something, like in `context/Auth.js`?
- https://github.com/Golf-Drill-Challenge-App/Golf-App/blob/main/context/Auth.js#L41

I set up the observer to only run when the user pressed the button as I was paranoid about it executing too often etc.

Reply:
You can pass it an empty array so it only executes once. Although that has failed me once with the debounce timer thingie...